sonic_y_cable/y_cable.py

```diff
@@ -0,0 +1,348 @@
+
+    Returns:
+        a Boolean, true if the toggle succeeded and false if it did not succeed.
+    """
+
+    buffer = bytearray([3])
+    curr_offset = Y_CABLE_SWITCH_MUX_DIRECTION
+
+    if platform_chassis is not None:
+        result = platform_chassis.get_sfp(
+            physical_port).write_eeprom(curr_offset, 1, buffer)
+    else:
+        helper_logger.log_error("platform_chassis is not loaded, failed to toggle mux to TOR B")
+        return False
+
+    return result
+
+
+def check_read_side(physical_port):
+    """
+    This API specifically checks which side of the Y cable the reads are actually getting performed
+    TOR A , TOR B or NIC (TOR A is chosed by convention as TOR 1 and TOR B is chosen as TOR 2 hereafter)
+    API basically reads 1 byte at upper page 4 offset 128 and checks which side the read is being performed
+    from.
+
+
+    Register Specification of upper page 0x4 at offset 128 is documented below
+
+    Byte offset   bits     Name                    Description
+                  7-3      Reserved                Determine which side of the cable you are reading from - specifically to differentiate TOR #1 and TOR #2:
+                                                   0b1 : Reading from indicated side, 0b0 not reading from that side.
+                  2        TOR #1 Side
+                  1        TOR #2 Side
+                  0        NIC Side
+    Args:
+         physical_port:
+             an Integer, the actual physical port connected to Y end of a Y cable which can which side reading the MUX from
+
+    Returns:
+        an Integer, 1 if reading the Y cable from TOR A side(TOR 1).
+                  , 2 if reading the Y cable from TOR B side(TOR 2).
+                  , 0 if reading the Y cable from NIC side.
+                  , -1 if reading the Y cable API fails.
+    """
+
+    curr_offset = Y_CABLE_DETERMINE_CABLE_READ_SIDE
+
+    if platform_chassis is not None:
+        result = platform_chassis.get_sfp(
+            physical_port).read_eeprom(curr_offset, 1)
+    else:
+        helper_logger.log_error("platform_chassis is not loaded, failed to check read side")
+        return -1
+
+    regval_read = struct.unpack(">i", result)
+
+    if ((regval_read >> 2) & 0x01):
+        helper_logger.log_info("Reading from TOR A")
+        return 1
+    elif ((regval_read >> 1) & 0x01):
+        helper_logger.log_info("Reading from TOR B")
+        return 2
+    elif (regval_read & 0x01):
+        helper_logger.log_info("Reading from NIC side")
+        return 0
+    else:
+        helper_logger.log_error(
+            "Error: unknown status for checking which side regval = {} ".format(result))
+        return -1
+
+    return -1
+
+
+def check_active_linked_tor_side(physical_port):
+    """
+    This API specifically ichecks which side of the Y cable is actively linked and Routing
+    TOR A , TOR B (TOR A is chosed by convention as TOR 1 and TOR B is chosen as TOR 2 hereafter)
```

Comment:
This is a bit confusing. Should probably just stick with "TOR A" and "TOR B" rather than changing terminology, plus you still do use "TOR A" and "TOR B" in the "Returns" section. Suggest sticking with "TOR A" and "TOR B" throughout the file.

Reply:
fixed. the doc string says TOR A and TOR B. 
kept the register spec as well, will be helpful for debugging/enhancement later on.